quesma/queryparser/pancake_transformer.go

```diff
@@ -297,21 +300,23 @@ func (a *pancakeTransformer) findParentBucketLayer(layers []*pancakeModelLayer,
 	return layer, nil
 }
 
-func (a *pancakeTransformer) createTopHitPancakes(pancake *pancakeModel) (result []*pancakeModel, err error) {
+func (a *pancakeTransformer) createTopHitAndTopMetricsPancakes(pancake *pancakeModel) (result []*pancakeModel, err error) {
 	for layerIdx, layer := range pancake.layers {
 		metricsWithoutTopHits := make([]*pancakeModelMetricAggregation, 0, len(layer.currentMetricAggregations))
-		for _, metric := range layer.currentMetricAggregations {
-			if _, isTopHits := metric.queryType.(metrics_aggregations.TopHits); isTopHits {
-				canOptimize := layerIdx == len(pancake.layers)-1 && len(layer.currentMetricAggregations) == 1
+		for metricIDx, metric := range layer.currentMetricAggregations {
+			switch metric.queryType.(type) {
+			case metrics_aggregations.TopMetrics, metrics_aggregations.TopHits:
+				canOptimize := layerIdx == len(pancake.layers)-1 && (len(layer.currentMetricAggregations) == 1 ||
```

Comment:
The condition is quite complex. Maybe we should add a comment with examples.

Reply:
Broke it down to named variables.